File: api/app/schemas/lineup.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class LineupEntryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    lineup_id: int
    roster_entry_id: int | None = None
    player_id: int
    player_name: str | None = None
    player_position: str | None = None
    player_school: str | None = None
    slot: str
    is_starter: bool


class LineupRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    league_id: int
    team_id: int
    season: int
    week: int
    status: str
    locked_at: datetime | None = None
    entries: list[LineupEntryRead]


class LineupAssignment(BaseModel):
    roster_entry_id: int | None = None
    player_id: int
    slot: str
    is_starter: bool


class LineupUpdateRequest(BaseModel):
    assignments: list[LineupAssignment]


class LineupUpdateResponse(BaseModel):
    data: LineupRead

File: web/client/types/lineup.ts
export interface LineupEntry {
  id: number;
  lineup_id: number;
  roster_entry_id: number | null;
  player_id: number;
  player_name?: string | null;
  player_position?: string | null;
  player_school?: string | null;
  slot: string;
  is_starter: boolean;
}

export interface Lineup {
  id: number;
  league_id: number;
  team_id: number;
  season: number;
  week: number;
  status: string;
  locked_at?: string | null;
  entries: LineupEntry[];
}

export interface LineupAssignment {
  roster_entry_id?: number | null;
  player_id: number;
  slot: string;
  is_starter: boolean;
}

export interface LineupUpdateRequest {
  assignments: LineupAssignment[];
}

export interface LineupUpdateResponse {
  data: Lineup;
}
